cmd/helm/lint.go

```diff
@@ -112,14 +122,17 @@ func newLintCmd(out io.Writer) *cobra.Command {
 			if failed > 0 {
 				return errors.New(summary)
 			}
-			fmt.Fprintln(out, summary)
+			if !client.Quiet {
+				fmt.Fprintln(out, summary)
+			}
 			return nil
 		},
 	}
 
 	f := cmd.Flags()
 	f.BoolVar(&client.Strict, "strict", false, "fail on lint warnings")
 	f.BoolVar(&client.WithSubcharts, "with-subcharts", false, "lint dependent charts")
+	f.BoolVar(&client.Quiet, "quiet", false, "print only warnings and errors")
```

Comment:
The message here is not correct. `--quiet` will only show errors by default. If you want warnings to display you need to also have `--strict` set. Was the intent to always show warnings when quiet? If so, there is a large code change needed. If the current functionality is ok then the flag doc string needs to be updated.

Reply:
Discussed this with @mattfarina last week and a good catch by him. Sorry only getting to respond now.

You can reproduce this by creating a scaffold chart (`helm create`) and remove the `templates/` directory. You can see the difference when you test with, without the flag and with `strict` flag:

```console
$ helm lint lint-warns
==> Linting lint-warns
[INFO] Chart.yaml: icon is recommended
[WARNING] templates/: directory not found

1 chart(s) linted, 0 chart(s) failed

$ helm lint lint-warns --quiet

$ helm lint lint-warns --quiet --strict 
==> Linting lint-warns
[WARNING] templates/: directory not found

Error: 1 chart(s) linted, 1 chart(s) failed
```

@presztak How do you think it should work with the `--quiet` flag?

Reply:
@hickeyma

I believe that intention of this issue was displaying errors as well warnings.
I fixed the code and it should be better right now. What do you think about this solution?
I also added new test which checks is quiet flag works well with chart which has only warnings.

If this solution is not ok we can turn back to the previous one and change quiet flag description.